packages/theme-check-common/src/checks/unique-static-block-id/index.ts

```diff
@@ -0,0 +1,64 @@
+import { LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+
+export const UniqueStaticBlockId: LiquidCheckDefinition = {
+  meta: {
+    code: 'UniqueStaticBlockId',
+    name: 'Prevent use of duplicated ids for blocks rendered statically',
+    docs: {
+      description:
+        'This check is aimed at preventing the use of duplicated ids for blocks rendered statically.',
+      url: 'https://shopify.dev/docs/themes/tools/theme-check/checks',
+      recommended: true,
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.ERROR,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    const usedIds: string[] = [];
+    return {
+      async LiquidTag(node) {
+        if (node.name !== 'content_for') {
+          return;
+        }
+
+        const [blockType] = node.markup.split(',');
+
+        if (blockType.replace(/["']/g, '') !== 'block') {
+          return;
+        }
+
+        const idRegex = /id: ["'](\S+)["']/;
+
+        const idValueMatch = idRegex.exec(node.markup);
+
+        if (idValueMatch == null) {
+          return;
+        }
+
+        const filteredIdValue = idValueMatch[1];
```

Comment:
nit: At this point in the code, you could destructure the two match elements out of `idValueMatch` to be a bit more concise when you use each value. 
Something like:
```
const [entireIdTerm, idValue] = idValueMatch
```